### DISCUSSION TITLE
Multi-platform app

### DISCUSSION THREAD

**Discussion by rzelnik** [alainm23/planify#654] [Ideas]
I tried many apps for task planning and project management. Most of them are too limited or too closed and therefore expensive. Planner is different: it is both advanced and open source. This makes it special. It would be great to have Planner available also for Windows and Mac users. I would like to recommend this app to my friends but most of them don't use Linux. What tools or processes are necessary for multi-platform releasing?

**Comment by koudi** [comment]
GTK apps should be somewhat portable, but as I understand Planner uses DBus and Evolution data server, which are very linux specific. Porting to windows would mean removing a lot of functionality or reimplementing it in portable way.

**Comment by OmarShawky1** [comment]
A very costy tweek is to use Xming in windows to open linux programs but generally this is not advised due to ram extensive and requires tedious work (for newbies)

**Comment by akvadrako** [comment]
There is an open-source Todoist app for Android: https://tisktasks.com/

**Reply by OakLD** [reply]
Nice it's OSS, but I'm sure that people in this thread are looking for a different kind of beast...

**Comment by benmordecai** [comment]
As a Todoist user, I've been looking for something that would be a good FOSS alternative, especially if I can self-host the data, but the use of Windows for work is out of my control, and the need for app support on Android is a necessity. Elementary Planner is almost perfect except for the fact that I can't use it cross platform, which prevents me from being able to replace Todoist. At that point if I am using Todoist anyway, I might as well use the official apps.

**Reply by ibadullahsyed** [reply]
its been 3 years now. have you found a good foss alternative that is crossplatform (using Webdav or Nextcloud) wuth good ui? pls tell me pls pls

**Reply by achmafooma** [reply]
> its been 3 years now. have you found a good foss alternative that is crossplatform (using Webdav or Nextcloud) wuth good ui? pls tell me pls pls

I'm using self-hosted [NextCloud](https://nextcloud.com/) for my CalDAV tasks/calendar server, Planify for managing tasks on Linux (and on Windows with my dev/in-progress build), Reminders on iOS and Mac (built-in Apple app), and [Tasks.org](https://tasks.org/) on Android. It's less integrated/consistent than a tool like Todoist, but it's working well for me.

I just wish the NextCloud web UI could deal with recurring tasks :-/ I have to avoid using the webapp because a lot of my tasks are recurring and it'll mess 'em up. But as long as I don't modify them in the web UI they still sync correctly with recurrence settings across the other apps.

**Reply by benmordecai** [reply]
I'm still using Todoist

**Comment by sramkrishna** [comment]
theoretically, you could run it under WSL2 in windows.

**Comment by achmafooma** [comment]
Planify _can_ be built for Windows, and seems to work as expected. I used the MSYS2 UCRT64 build environment.

I'm considering (but not promising yet!) contributing the necessary code and documentation for Windows support, either officially or as a soft-fork. I just opened a discussion/poll here: https://github.com/alainm23/planify/discussions/2105
